docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -3118,9 +3119,33 @@ error rate is not very intuitive for the end user.
 At some point in the future, we might make this more user friendly. However, for the 
 sake of a minimum functional model, this is satisfactory. 
 
-One function has an additional parameter. The typographic function needs an input for 
-the probability that a corrupted token contains the original token. For simplicity, 
-we will use 0.1 and the user will not be able to edit this value. 
+A few functions have additional parameters that allow the user more control. These 
+inputs will not be specified in the main settings for pseudopeople listed above. We 
+imagine that a yaml file (or similar file type) will be available for users to download 
```

Comment:
I was imagining the interface would be through python and R functions in a module called pseudopeople:
```
import psuedopeople as pp
pp.simulate_decennial_data(first_nickname_pr=.2)
```

I guess this might get unwieldly with the number of options we are heading towards, so I'm not set on it.  But I wanted to call it out, in contrast to the specification-file approach you are proposing.

Reply:
We could do something like this! This is similar to what I imagined above for the "basic" parameters. I have a hunch the list of optional/additional parameters might expand over time and having them in a separate file might make it cleaner. But we can definitely adjust this once we have a better picture of what pseudopeople will look like.

Reply:
What do we gain by separating the parameters users can pass into "basic" and "additional" parameters? It seems like it would be simpler to me to have a single set of parameters which can be specified via config file, Python dictionary/R list, or directly as function arguments as in Abie's example.

Reply:
There are a lot of parameters right now, and being realistic I expect the list to expand as these functions are further built out (software used, blocking requirements, confidence level needed, etc.). Just with the current "basic" options, I counted 58 parameters for the census. 

I thought there would be value in starting simpler and having some "basic" options for a user who is new or has a more simplistic need. Then "additional" parameters for users with more specific needs and who is not intimidated by the number of inputs! It also sounds to me like we should have a conversation about the user interface soon.

Reply:
As @zmbc says, I was imagining a single configuration file that includes all parameters -- there would be no distinction between "basic" and "additional" in the full parameter specification. I was imagining the config file would get read into some in-memory data structure like a Python dictionary, which would then get passed to a top level function that adds noise to the dataset (so if desired, the user could instead specify parameters directly in a dictionary instead of in a file).

The reason we were talking about "basic" vs. "additional" is because, for the sake of simplicity, we had all previously agreed to have the user specify exactly two parameters for each column (the "overall fraction of rows with noise," and the "amount of noise within each corrupted row," or something along those lines -- except that the second of these doesn't make sense for some types of noise such as incorrect selection), and have any additional parameters be implementation details.

But there are a **lot** of parameters, and the list will probably grow as we improve the noise functions. I think it could be useful for users to be able to tweak the different parameters to see how variations in noise affect the linking, and as @SylLutze says, we probably (I think?) want to have a simplified set of parameters (e.g., the two "basic" ones we agreed to previously) with reasonable defaults for everything, so as not to overwhelm a new user. 

As for @aflaxman's suggestion, it seems to me like having a function with 58 or 100 parameters would be rather unwieldy, which is why I suggested a configuration file (or dictionary). But I guess if we *really* want to have all the parameters as arguments in some top-level function, I guess we could... although that would be harder to maintain if we ever change the underlying noise functions.

Reply:
> it seems to me like having a function with 58 or 100 parameters would be rather unwieldy

We could just have the function accept `**kwargs` and treat that similar to the config file/data structure. Only downside is the lack of nesting but we could figure out an approach for that too (e.g. `{observer}_{column}_{param}` or something).

> we probably (I think?) want to have a simplified set of parameters (e.g., the two "basic" ones we agreed to previously) with reasonable defaults for everything

My vote would be to have reasonable defaults for **everything** -- the user can run pseudopeople without specifying anything regarding noise. Then, all those parameters can be overridden. Separating into "basic" and "additional" seems like more of a documentation question to me, rather than a distinction we need to make in code.

Reply:
> My vote would be to have reasonable defaults for **everything**

👍 

Regarding how we go beyond the defaults, I find this discussion very helpful, but I'm not sure what will work best.

Reply:
For reference, [here](https://github.com/ihmeuw/vivarium_research_prl/blob/main/src/vivarium_research_prl/find_kids/noisify_data.py) is the code where I'm actually adding noise to the census and WIC data, which is essentially a hard-coded version of what I imagined a config file would look like.

And [here](https://github.com/ihmeuw/vivarium_research_prl/blob/main/src/vivarium_research_prl/noise/noisify.py) are some functions I started to write to parse the input parameters and add noise in the correct way (the first of these gets called repeatedly in the module above; the second isn't used yet).

If we went with Zeb's suggestion of passing `**kwargs`, my current approach would require additional nesting, e.g., `{observer}_{column}_{noise_function}_{param}`, and you would have to simultaneously pass `row_fraction=0.01`, for example. But maybe we could come up with a way to simplify that? Maybe `obeserver`, `column`, and `row_fraction` should all just be separate required arguments because those are needed for everything, and then the noise is specified by `{noise_function}_{param}`. Or `{noise_function}` and `{param}` could even be split into two arguments. Anyway...

I agree that we should come up with reasonable defaults for everything, and that it's not necessary to make a distinction between "basic" and "additional" for the code, so we don't need to specify that for the engineers.

As for what we should tell the engineers to implement, it seems to me like the configuration file / nested dictionary approach is the easiest and most flexible way to start, and if we decide we want to put a nicer wrapper on the user interface such as being able to pass keyword arguments, that's something we could add on later. What do you think, @aflaxman and @zmbc?

Reply:
> it seems to me like the configuration file / nested dictionary approach is the easiest and most flexible way to start

Agree, I think we can add Abie's version later pretty easily.